src/main/java/domain/AttendanceBook.java

```diff
@@ -0,0 +1,118 @@
+package domain;
+
+import exception.AttendanceNotExistException;
+import exception.DuplicateAttendanceException;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.*;
+
+public class AttendanceBook {
+    private final HashSet<Attendance> attendances; // key: 몇 일, value: 출석 시간
+
+    public AttendanceBook(int year, int month) {
+        Month customMonth = Month.of(month);
+
+        attendances = new HashSet<>();
+        for (int day = 1; day <= customMonth.getLastDay(); day++) {
+            if (!customMonth.isHoliday(day)) {
+                LocalDate date = LocalDate.of(year, month, day);
+                attendances.add(Attendance.empty(date));
+            }
+        }
+    }
+
+    public Attendance create(LocalDate date, LocalTime time) {
+        final boolean isExist = attendances.stream()
+                .anyMatch(attendance -> attendance.getDate().isEqual(date) && !attendance.isEmpty());
+        if (isExist) {
+            throw new DuplicateAttendanceException();
+        }
+        Attendance attendance = Attendance.of(date, time);
+        addAttendance(attendance);
+        return attendance;
+    }
+
+    public Attendance findAttendanceByDate(LocalDate date) {
+        return attendances.stream()
+                .filter(attendance -> attendance.getDate().isEqual(date))
+                .findFirst()
+                .orElse(Attendance.empty(date));
+    }
+
+    public void replace(LocalDate date, LocalTime time) {
+        Attendance oldAttendance = attendances.stream()
+                .filter(attendance -> attendance.getDate().isEqual(date))
+                .findFirst()
+                .orElseThrow(AttendanceNotExistException::new);
+        Attendance modifiedAttendance = oldAttendance.modify(time);
+        addAttendance(modifiedAttendance);
+    }
+
+    public List<Attendance> getAllAttendances(final int limitDay) {
+        return attendances.stream()
+                .sorted(Comparator.comparing(Attendance::getDate))
+                .filter(attendance -> attendance.getDate().getDayOfMonth() < limitDay)
+                .toList();
+    }
+
+    public Map<AttendanceStatus, Integer> calculateAttendanceResult(final int limitDay) {
+        Map<AttendanceStatus, Integer> result = new HashMap<>();
+        result.put(AttendanceStatus.ATTENDANCE, 0);
+        result.put(AttendanceStatus.LATE, 0);
+        result.put(AttendanceStatus.ABSENCE, 0);
+
+        List<Attendance> attendances = getAllAttendances(limitDay);
+        for (Attendance attendance : attendances) {
+            AttendanceStatus status = attendance.getStatus();
+            if (status == AttendanceStatus.TRUANCY) {
+                status = AttendanceStatus.ABSENCE;
+            }
+            result.replace(status, result.get(status) + 1);
+        }
+        return result;
```

Comment:
```suggestion
        Map<AttendanceStatus, Integer> result = new HashMap<>();

        List<Attendance> attendances = getAllAttendances(limitDay);
        for (Attendance attendance : attendances) {
            AttendanceStatus status = attendance.getStatus();
            if (status == AttendanceStatus.TRUANCY) {
                status = AttendanceStatus.ABSENCE;
            }
            int newValue = result.getOrDefault(status, 0) + 1;
            result.put(status, newValue);
        }
        return result;
```

getOrDefault를 사용해 초기화 코드를 없애는 것도 방법이 될 것 같습니다.

Reply:
확인해주셔서 감사합니다! 내용 반영했습니다.

[fix(AttendanceBook): getOrDefault를 사용해서 불필요한 코드 삭제](https://github.com/woowacourse/java-attendance/pull/42/commits/ae2d47d8f8f97932445a53c23887feb7ed502e78)